src/workload.rs

```diff
@@ -78,12 +82,32 @@ impl TryFrom<Option<xds::istio::workload::WorkloadStatus>> for HealthStatus {
     }
 }
 
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Default, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct GatewayAddress {
+    #[serde(default)]
+    pub address: Option<gatewayaddress::Address>,
+    #[serde(default)]
+    pub port: u16,
+}
+
+pub mod gatewayaddress {
+    #[derive(Debug, Hash, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
+    #[serde(tag = "address", content = "content")]
+    pub enum Address {
+        Hostname(String),
+        IP(std::net::IpAddr),
+    }
+}
+
 #[derive(Debug, Hash, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct Workload {
     pub workload_ip: IpAddr,
     #[serde(default)]
-    pub waypoint_addresses: Vec<IpAddr>,
+    pub waypoint: GatewayAddress,
+    #[serde(default)]
+    pub network_gateway: GatewayAddress,
```

Comment:
```suggestion
    pub waypoint: Option<GatewayAddress>,
    #[serde(default)]
    pub network_gateway: Option<GatewayAddress>,
```

Then remove the Option<> inside of GatewayAddress.

We should either allow having an ip/host+port, or not, which is better represented with above

Reply:
implemented in https://github.com/istio/ztunnel/pull/486/commits/6d7827f3110d69e6b1eb94d5975120c92fa3cf79